src/gtest/test_joinsplit.cpp

```diff
@@ -587,3 +587,82 @@ TEST(Joinsplit, NoteClass)
     ASSERT_EQ(note.r, clone.r);
     ASSERT_EQ(note.a_pk, clone.a_pk);
 }
+
+TEST(Joinsplit, BasicJoinsplitVerification)
+{
+    LoadProofParameters();
+
+    // We only check that joinsplits are constructed properly
+    // and verify them here.
+    //
+    // See #471, #520, #459 and probably others.
+    //
+    // There may be ways to use boost tests to catch failing
+    // threads or processes (?) but they appear to not work
+    // on all platforms and would gently push us down an ugly
+    // path. We should just fix the assertions.
+    //
+    // Also, it's generally libzcash's job to ensure the
+    // integrity of the scheme through its own tests.
+
+    // construct a merkle tree
+    SproutMerkleTree merkleTree;
+
+    auto k = libzcash::SproutSpendingKey::random();
+    auto addr = k.address();
+
+    libzcash::SproutNote note(addr.a_pk, 100, uint256(), uint256());
+
+    // commitment from coin
```

Comment:
```suggestion
    // commitment from note
```
Another sign of how old this code is :-)